### PR TITLE
Server refactor

### PR DESCRIPTION
**Pull request by kronicker**
- Refactored helpers
- Returning data flattened and its naming unified
- Description sanitized for safety
- Changes applied to client code making it more readable and robust

### CODE REVIEW

File: client/main.js [hunk 1]
```diff
@@ -7,36 +7,32 @@ import format from 'date-fns/format';
 import hrLocale from 'date-fns/locale/hr'
 
 function Event({ event }) {
+  const { category, title, date, time, location, description, image, link } = event;
+  const Description = desc => ({ __html: desc });
   const EventLink = ({ link }) => link ? <a href={ link.url }>{ link.label }</a> : null;
   const EventTime = ({ time }) => time ? <span>@{ time } h</span> : null;
```
Comment (vladimyr) [inline client/main.js]: One could argue that those two are now pretty generic and should be named `Link` and `Time` respectfully? 🤔 
Also they should probably end up being pulled outside from `Event` component...

File: client/main.js [hunk 2]
```diff
@@ -7,36 +7,32 @@ import format from 'date-fns/format';
...
-          <EventTime time={event.time}/>
+          <EventTime time={time}/>
         </h4>
-        <span className="label label-primary">{ eventCategory }</span>
-      </p>
-      <p dangerouslySetInnerHTML={ eventDescription(event.description) }></p>
-      <EventLink link={ event.link } />
+        <span className="label label-primary">{ category }</span>
+      </div>
+      <p dangerouslySetInnerHTML={ Description(description) }></p>
```
Comment (vladimyr) [inline client/main.js]: Wait, wat? You have component that returns html only or I'm missing something?

File: server/calendar/helpers.js [hunk 3]
```diff
@@ -10,55 +11,72 @@ const reUrl = /url\((.*?)\)/;
...
 }
 
 function readPhotoUrl($photo) {
   if ($photo.length <= 0) return;
 
-  let style = $photo.attr('style');
-  let match = style.match(reUrl);
+  const match = $photo
+    .attr('style')
+    .match(reUrl);
```
Comment (vladimyr) [inline server/calendar/helpers.js]: Not big fan of jQuery-esque formatting. If you really want to narrow it down make it oneliner.

File: server/calendar/helpers.js [hunk 4]
```diff
@@ -10,55 +11,72 @@ const reUrl = /url\((.*?)\)/;
...
 }
+
+function sanitizeHTML(data) {
+  return sanitizeHtml(data, {
+    textFilter(text) {
+      return text.replace(/\n/g, '<br>')
+    }
+  })
+}
+
```
Comment (vladimyr) [inline server/calendar/helpers.js]: How about:
```js
const textFilter = text => text.replace(/\n/g, '<br>');
const sanitizeHtml = html => require('sanitize-html')(html, { textFilter }); 
```

File: server/calendar/scraper.js [hunk 5]
```diff
@@ -1,47 +1,58 @@
...
+    category: info[HR_LOCALES.category],
+    title,
+    date: info[HR_LOCALES.date],
+    time,
+    location: info[HR_LOCALES.location],
+    ticket: info[HR_LOCALES.ticket],
+    description,
+    image,
+    link
+  };
```
Comment (vladimyr) [inline server/calendar/scraper.js]: Well, you ended up with table so populate it fully by inlining those 5 remaining vars? :bowtie: 
Or move those 4 up?

File: server/calendar/scraper.js [hunk 6]
```diff
@@ -1,47 +1,58 @@
...
+  readPhotoUrl, readLink, sanitizeHTML
 } = require('./helpers.js');
 
-module.exports = function ($, $el) {
+const HR_LOCALES = {
+  location: 'Lokacija',
+  date: 'Datum',
+  ticket: 'Ulaznica',
+  category: 'Kategorija'
+}
```
Comment (vladimyr) [inline server/calendar/scraper.js:14]: Nice 😄

**Review by vladimyr — CHANGES_REQUESTED** [review]
Thx for changing all my `let`s to `const`s 😉

File: client/main.js (hunk 2 above)
Comment (kronicker) [reply client/main.js]: Actually I have mental illness for writing this code. I made a totally unnecessary function (with Pascal case in the middle of nowhere 👨 🔫 ) that returns object. It should have been:
```js
<p dangerouslySetInnerHTML={{_html: event.description}}></p>
```

Comment (kronicker) [reply client/main.js]: Only answer that comes to mind is waaay too much ☕️

File: server/calendar/helpers.js (hunk 4 above)
Comment (kronicker) [reply server/calendar/helpers.js]: 💯 🥇

File: server/calendar/helpers.js [hunk 7]
```diff
@@ -10,55 +11,72 @@ const reUrl = /url\((.*?)\)/;
...
+  const day = date.getDate();
+  const month = date.getMonth() + 1;
+  const year = date.getFullYear();
+  return `${ month }/${ day }/${ year }`;
+}
+
+function normalizeDateString(date) {
+  const dateChunks = date.split('/');
+  dateChunks.splice(0,0, dateChunks.splice(1,1)[0]);
+  return dateChunks.join('/');
```
[the remark this answers is not in the document]
Comment (kronicker) [reply server/calendar/helpers.js]: Fine, I'll refactor it. But you're killing its magic spark ❇️ ✨ 
True magic

**Comment by kronicker** [comment]
Sure! Anytime! If you need more info on the topic you can read this [article](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75).
🤣

File: server/calendar/helpers.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (vladimyr) [reply server/calendar/helpers.js]: I was tempted to run it in node REPL but I resisted. :blush:

**Comment by vladimyr** [comment]
>Sure! Anytime! If you need more info on the topic you can read this article.
:rofl:

I'm sure you somehow unintentionally typed wrong link, you surely meant this [article](http://wesbos.com/let-vs-const/) instead :trollface:

File: server/calendar/helpers.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (kronicker) [reply server/calendar/helpers.js]: You should have! It's that good! 😝 
Can't touch this

File: server/calendar/helpers.js (hunk 4 above)
Comment (kronicker) [reply server/calendar/helpers.js]: Though I'm not a fan of requiring in the middle of the file if not necessary...

Comment (vladimyr) [reply server/calendar/helpers.js]: Ofc you'll pull it up. `const` helpers should be listed top most (after imports) due to Temporal Dead Zone 😉

**Comment by vladimyr** [comment]
>c2207bb - Kill your own spirit

👌

File: client/main.js [hunk 8]
```diff
@@ -6,37 +6,34 @@ import http from 'axios';
 import format from 'date-fns/format';
 import hrLocale from 'date-fns/locale/hr'
 
+const Link = ({ link }) => <a href={ link.url }>{ link.label }</a>;
+
+const Time = ({ time }) => <span>@{ time } h</span>;
```
Comment (vladimyr) [inline client/main.js]: Blank line, well, if it makes you happy 😄

File: client/main.js [hunk 9]
```diff
@@ -6,37 +6,34 @@ import http from 'axios';
...
+    <li className="event list-group-item">
+      <h2>{ title }</h2>
+      <div>
         <h4>
-          { event.info && event.info.Lokacija ? event.info.Lokacija : null }
+          { location }
           &nbsp;
-          <EventTime time={event.time}/>
+          { time && <Time time={ time }/>}
         </h4>
```
Comment (vladimyr) [inline client/main.js]: If I were you, I would wrap `location` with `span.location` and used `h3` to preserve semantics 😉

File: client/main.js [hunk 10]
```diff
@@ -6,37 +6,34 @@ import http from 'axios';
...
-        <span className="label label-primary">{ eventCategory }</span>
-      </p>
-      <p dangerouslySetInnerHTML={ eventDescription(event.description) }></p>
-      <EventLink link={ event.link } />
+        <span className="label label-primary">{ category }</span>
+      </div>
+      <p dangerouslySetInnerHTML={{ __html: desc }}></p>
+      { link && <Link link={ link }/>}
```
Comment (vladimyr) [inline client/main.js]: 😻

File: server/calendar/helpers.js (hunk 4 above)
Comment (kronicker) [reply server/calendar/helpers.js]: 👍  never looked into that actually so 🙇

File: server/calendar/helpers.js [hunk 11]
```diff
@@ -2,63 +2,76 @@
...
+  const html = entities.decode($info.html());
+  return html
+    .trim()
+    .split('<br>')
+    .filter(r => r.length > 0)
+    .reduce((info, record) => {
+      const [key, val] = record.split(/:\s*/);
+      info[key] = val;
+      return info
+    }, {});
```
Comment (vladimyr) [inline server/calendar/helpers.js:43]: Well, technically you could turn it all into one big chain 🚂 🚋 🚋 🚋  😁

File: server/calendar/helpers.js [hunk 12]
```diff
@@ -2,63 +2,76 @@
...
+  const month = date.getMonth() + 1;
+  const year = date.getFullYear();
+  return `${ month }/${ day }/${ year }`;
+}
+
+function normalizeHrDateString(date) {
+  const dateChunks = date.split('/');
+  const day = dateChunks[0];
+  const month = dateChunks[1];
+  const year = dateChunks[2];
```
Comment (vladimyr) [inline server/calendar/helpers.js]: How 'bout
```js
 const [day, month, year] = date.split('/');
```
No chunks, no cry 😁

File: server/calendar/scraper.js [hunk 13]
```diff
@@ -1,47 +1,51 @@
...
+  const event = {};
+  event.category = info[HR_LOCALES.category];
+  event.title = $event.find('h2').text();
+  event.date = info[HR_LOCALES.date];
+  event.time = readTime($el.find('.dan'));
+  event.location = info[HR_LOCALES.location];
+  event.description = sanitizeHTML($content.eq(1).text());
+  event.ticket = info[HR_LOCALES.ticket];
+  event.image = readPhotoUrl($event.find('.foto'));
+  event.link = readLink($content.eq(1).next('a'));
```
Comment (vladimyr) [inline server/calendar/scraper.js]: Why not `Object.assign`? 😕

File: server/calendar/helpers.js [hunk 14]
```diff
@@ -2,63 +2,76 @@
...
   if ($photo.length <= 0) return;
 
-  let style = $photo.attr('style');
-  let match = style.match(reUrl);
+  const match = $photo.attr('style').match(reUrl);
   if (!match || !match[1]) return;
 
-  let path = unquote(match[1].trim());
+  const path = unquote(match[1].trim());
   return url.resolve(baseUrl, path);
```
Comment (vladimyr) [inline server/calendar/helpers.js]: Let me rewrite it quickly:
```js
const [match] = $photo.attr('style').match(reUrl) || [];
if (!match) return;
const path = unquote(match.trim());
return url.resolve(baseUrl, path);
```

File: server/calendar/helpers.js [hunk 15]
```diff
@@ -2,63 +2,76 @@
...
-  readInfo, readPhotoUrl, readLink
+  readDate, readTime, readInfo,
+  readPhotoUrl, readLink, sanitizeHTML
 };
 
 function readDate($date) {
   let date = $date.text().trim();
-  if (reDate.test(date)) return date;
-  return formatDate(new Date());
+  return reDate.test(date) ? normalizeHrDateString(date) : formatDate(new Date());
```
Comment (vladimyr) [inline server/calendar/helpers.js]: Still looks bit clumsy 😢

Comment (kronicker) [reply server/calendar/helpers.js]: That's why we should use momentjs...
It has the [feature we need](http://momentjs.com/docs/#/parsing/string-format/).

File: server/calendar/helpers.js (hunk 11 above)
Comment (kronicker) [reply server/calendar/helpers.js:43]: Train

File: server/calendar/helpers.js (hunk 14 above)
Comment (kronicker) [reply server/calendar/helpers.js]: Now I'm tempted to do:
```js
const [match] = $photo.attr('style').match(reUrl) || [];
return match ? url.resolve(baseUrl, unquote(match.trim()) : '';
```
JK 😆

File: server/calendar/scraper.js (hunk 13 above)
Comment (kronicker) [reply server/calendar/scraper.js]: Yeah, why not? 😉

**Pull request merged**